File: 0836-rectangle-overlap/0836-rectangle-overlap.ts
function isRectangleOverlap(rec1: number[], rec2: number[]): boolean {
    const left = Math.max(rec1[0], rec2[0]);
    const right = Math.min(rec1[2], rec2[2]);
    const bottom = Math.max(rec1[1], rec2[1]);
    const top = Math.min(rec1[3], rec2[3]);
    return left < right && bottom < top;
}

File: 0836-rectangle-overlap/0836-rectangle-overlap.rs
impl Solution {
    pub fn is_rectangle_overlap(rec1: Vec<i32>, rec2: Vec<i32>) -> bool {
        let left   = rec1[0].max(rec2[0]);
        let right  = rec1[2].min(rec2[2]);
        let bottom = rec1[1].max(rec2[1]);
        let top    = rec1[3].min(rec2[3]);
        left < right && bottom < top
    }
}